Tests/SDFCoreTests/MacroInitializerSorterTests.swift

```diff
@@ -1,201 +1,290 @@
-    let expected = """
-        init(d: String,
-             c: Int) {}
-    """
-    try await #expect(sut.run(onContent: content.rawValue).contains(expected))
-  }
-  
-  @Test(
-    "With a type that uses the macro inside of another type. It sorts the inner init parameters",
-    arguments: TestData.WithNestedDefinitionMacroInside.allCases)
-  func run_doesSortInnerWhenNested(content: TestData.WithNestedDefinitionMacroInside) async throws {
-    let expected = "    init(c: Int, d: String) {}"
-    try await #expect(sut.run(onContent: content.rawValue).contains(expected))
   }
-  
-  @Test(
-    "With a type that uses the macro inside of another type. It does not sort the outer init parameters",
-    arguments: TestData.WithNestedDefinitionMacroInside.allCases)
-  func run_doesNotSortOuterNested(content: TestData.WithNestedDefinitionMacroInside) async throws {
-    let expected = "  init(b: String, a: Int) {}"
-    try await #expect(sut.run(onContent: content.rawValue).contains(expected))
+
+  struct AssignmentTests {
+    let sut = MacroInitializerSorter(macroName: "TestMacro")
+
+    @Test(
+      "With a self assignments in the intitializer. It sorts them",
+      arguments: TestData.types
+    )
+    func run_sortsAssignments(type: String) async throws {
+      let content = """
+        @TestMacro
+        \(type) Some {
+          init(a: Int, b: String) {
+            self.b = b
+            self.a = a
+          }
+        }
+        """
+      let expected = """
+            self.a = a
+            self.b = b
+        """
+      #expect(try await sut.run(onContent: content).contains(expected))
+    }
+
+    @Test(
+      "With self assignments and other values in the intitializer. It sorts them with all the self assignments on top",
+      arguments: TestData.types
+    )
+    func run_sortsAssignmentsWithOther(type: String) async throws {
+      let content = """
+        @TestMacro
+        \(type) Some {
+          init(a: Int) {
+            b = Type()
+            self.a = a
+          }
+        }
+        """
+      let expected = """
+            self.a = a
+            b = Type()
+        """
+      #expect(try await sut.run(onContent: content).contains(expected))
+    }
+
+    @Test(
+      "With self assignments and block statements in the intitializer. It sorts them with all the self assignments on top",
+      arguments: TestData.types
+    )
+    func run_sortsAssignmentsWithBlockStatements(type: String) async throws {
+      let content = """
+        @TestMacro
+        \(type) Some {
+          init(a: Int) {
+            b = Type { some in
+              some.doSomething()
+            }
+            self.a = a
+          }
+        }
+        """
+      let expected = """
+            self.a = a
+            b = Type { some in
+              some.doSomething()
+            }
+        """
+      #expect(try await sut.run(onContent: content).contains(expected))
+    }
```

Comment:
we should have a test that looks like:

```suggestion
    }
    
    @Test(
      "With self assignments and block statements in the intitializer, on different sides of a super.init call. It sorts calls on either side of the super.init call without mixing them",
      arguments: TestData.types
    )
    func run_sortsAssignmentsWithBlockStatements(type: String) async throws {
      let content = """
        @TestMacro
        \(type) Some {
          init(a: Int) {
            c = Type { some in
              some.doSomething()
            }
            self.b = b
            let a = "test"
            super.init()
            self.d = d
            self.a = a
          }
        }
        """
      let expected = """
            self.b = b
            c = Type { some in
              some.doSomething()
            }
            let a = "test"
            super.init()
            self.a = a
            self.d = d
        """
      #expect(try await sut.run(onContent: content).contains(expected))
    }
```

We should not be moving assignment to a different side of a `super.init` method

Reply:
Good call!

Reply:
Note that the test case above is incorrect (we discussed this on messages but leaving here for posterity).

The real expectation should keep the "c = " line on top, so it'd become:

```swift
      let expected = """
            c = Type { some in
              some.doSomething()
            }
            self.b = b
            let a = "test"
            super.init()
            self.a = a
            self.d = d
        """
```